aten/src/ATen/native/Fill.cpp

```diff
@@ -55,7 +55,8 @@ Tensor& fill_quantized_(Tensor& self, const Scalar& value) {
 
 Tensor& fill_(Tensor& self, const Tensor& value) {
   TORCH_CHECK(value.dim() == 0, "fill_ only supports 0-dimension value tensor but got tensor with ", value.dim(), " dimensions.");
-  return fill_out(self, value.item());
+  self.copy_(value);
```

Comment:
mm, no? copy_ is synchronizing most of the time (when `value` is on the host and `self` on the device)

Reply:
so should I update this to only copy when value is on device? otherwise call fill_out?

Reply:
Yes, that'll give you best of both worlds